dig.go

```diff
@@ -346,15 +350,16 @@ func newNode(provides reflect.Type, ctor interface{}, ctype reflect.Type) (node,
 		ctor:     ctor,
 		ctype:    ctype,
 		deps:     deps,
+		key:      key(provides),
```

Comment:
hm, this looks wrong. `newNode` should take in the key which is being constructed, not derive one itself